reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClientConnect.java

```diff
@@ -214,13 +214,13 @@ public void subscribe(CoreSubscriber<? super Connection> actual) {
 				if (handler.toURI.isSecure()) {
 					if (_config.sslProvider == null) {
 						_config = new HttpClientConfig(config);
-						if (_config.checkProtocol(HttpClientConfig.h2c) && _config.protocols.length > 1) {
-							removeIncompatibleProtocol(_config, HttpProtocol.H2C);
-						}
 						_config.sslProvider = HttpClientSecure.defaultSslProvider(_config);
 					}
 
-					if (_config.checkProtocol(HttpClientConfig.h2c)) {
+					if (_config.checkProtocol(HttpClientConfig.h2c) && _config.protocols.length > 1) {
+						removeIncompatibleProtocol(_config, HttpProtocol.H2C);
+					}
+					else if (_config.checkProtocol(HttpClientConfig.h2c)) {
```

Comment:
What do you think about checking once for `_config.checkProtocol(HttpClientConfig.h2c)`?

Reply:
Sounds good.